### PR TITLE
feat: support multiple select table

### PR DESCRIPTION
**Pull request by CorrectRoadH**
## What does the PR do?
add a multiple select to select multiple tables as propmpt

CleanShot 2023-05-12 at 12 27 33

## Some ideas
1. Just a Select or a Select +  some buttons(for instance: select all) ?
I think having some buttons is necessary🤔 a database may contain many tables. If only to select the table manually. It is too much trouble.

2. Where is the best place for the button?
👀 put the button under the Select or above option? 
I think the latter is consistent with the design style. 

3. One button or Two buttons?
I think one button to select all(when selecting nothing) and to empty select(when having any select) is very simple for the user.

5. Empty select denotes all tables
It doesn't make sense if there is no table selected. it means not sending any table struct as the prompt. I think it is meaningless. So I  think it is a good design to send all tables when no selected denotes, equivalent select all tables.

### CODE REVIEW

File: src/components/ConversationView/index.tsx [hunk 1]
```diff
@@ -178,15 +178,17 @@ const ConversationView = () => {
...
-          const table = tables.find((table) => {
-            return table.name === currentConversation.tableName;
+        // Empty table name(such as []) denote all table. [] and `undefined` both are false in `if`
+        if (currentConversation.selectedTablesName) {
+          currentConversation.selectedTablesName.forEach((tableName) => {
+            const table = tables.find((table) => table.name === tableName);
+            if (table) {
+              if (tokens < MAX_TOKENS / 2) {
+                tokens += countTextTokens(schema + table.structure);
+                schema += table.structure;
```
Comment (tianzhou) [inline src/components/ConversationView/index.tsx:188]: Let's refactor it a bit.

1. first compile the table list based on whether there is selectedTablesName

2. count the token and append the table structure to the schema

**Review by tianzhou — CHANGES_REQUESTED** [review]

File: src/components/ConversationView/index.tsx (hunk 1 above)
Comment (CorrectRoadH) [reply src/components/ConversationView/index.tsx:188]: what does "compile the table list" mean?👀

Comment (tianzhou) [reply src/components/ConversationView/index.tsx:188]: Build the table list

**Comment by CorrectRoadH** [comment]
@tianzhou PTAL

**Review by tianzhou — APPROVED** [review]
LGTM. This is a good improvement!

**Pull request merged**